From 9ff2c68569141e820816ac2aa0b2ef99a856abdc Mon Sep 17 00:00:00 2001
Message-Id: <9ff2c68569141e820816ac2aa0b2ef99a856abdc.1443127809.git.yunwei.zhang@intel.com>
In-Reply-To: <e1b2a1693f0b49269998c391abb456b0391b67de.1443127809.git.yunwei.zhang@intel.com>
References: <e1b2a1693f0b49269998c391abb456b0391b67de.1443127809.git.yunwei.zhang@intel.com>
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Fri, 7 Aug 2015 15:14:30 +0530
Subject: [PATCH 9/9] FROM_UPSTREAM [VPG]: drm/i915: fix checksum write for
 automated test reply

DP spec requires the checksum of the last block read to be written
when replying to TEST_EDID_READ. This patch fixes the current code
to do the same.

v2: removed loop for jumping blocks and performed direct addition
as recommended by Daniel

Change-Id: I45c60d82cec85986c5588a97740f51339a1a0729
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-17498
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-by: Sonika Jindal <sonika.jindal@intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
---
 drivers/gpu/drm/i915/intel_dp.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index d8bce26..5abcb7f 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -4416,9 +4416,16 @@ static uint8_t intel_dp_autotest_edid(struct intel_dp *intel_dp)
 				      intel_dp->aux.i2c_defer_count);
 		intel_dp->compliance_test_data = INTEL_DP_RESOLUTION_FAILSAFE;
 	} else {
+		struct edid *block = intel_connector->edid;
+
+		/* We have to write the checksum
+		 * of the last block read
+		 */
+		block += intel_connector->edid->extensions;
+
 		if (!drm_dp_dpcd_write(&intel_dp->aux,
 					DP_TEST_EDID_CHECKSUM,
-					&intel_connector->edid->checksum,
+					&block->checksum,
 					1));
 			DRM_DEBUG_KMS("Failed to write EDID checksum\n");
 
-- 
1.9.1

